site/docs/view-spec.md

```diff
@@ -0,0 +1,254 @@
+
+Each ‘CREATE OR REPLACE VIEW’ statement creates a new view version metadata file for that view. The view version metadata file has following fields:
+
+
+| Required/Optional | Field Name | Description |
+|-------------------|------------|-------------|
+| Required | format-version | Json format version number for the view metadata spec. The view metadata spec and the corresponding format-version is independent of table spec. Starts with 1 and is incremented when there is a breaking change to view metadata. |
+| Required | object-type    | String identifying the type of object this metadata file is for. ‘View’ for all objects covered in this spec. |
+| Required | location | Location of the view metadata files |
+| Required | current-version-id | Current / latest version of the view. Set to ‘1’ when the view is first created. |
+| Optional | properties | A string to string map of view properties. Contains pre-set properties such as ‘comment’ describing the view, does not contain arbitrary metadata. |
+| Required | versions | An array of structs describing the last few versions of the view. Controlled by the table property: “version.history.num_entries”. See more below. |
+| Required | version-log | An array of structs describing the log of created versions. See more below. |
+| Optional | schemas | A list of schemas and alias-comment maps stored as objects with schema-id. See more below |
+| Optional | current-schema-id | ID of the current schema of the view |
+
+
+‘schemas’ is an array of structs with a field as shown below:
+
+| Required/Optional | Field Name | Description |
+|-------------------|------------|-------------|
+| Optional | schema | An array of structs describing the schema of the view. This field is the same as the ‘schema’ field from Iceberg table spec. See more below. |
+
+
+‘Versions’ is an array of structs with fields as shown below:
+
+| Required/Optional | Field Name | Description |
+|-------------------|------------|-------------|
+| Required | version-id | Monotonically increasing id indicating the version of the view. Starts with “1”. |
+| Required | timestamp-ms | Timestamp expressed in ms since epoch at which the version of the view was created. |
+| Required | summary | A string to string map of view properties to track version metadata. This field can be used by engines to store any necessary properties. Two currently required properties are described below. |
+| Required | sql | A string representing SQL definition of the view as input |
+| Required | dialect | A string specifying the dialect of the ‘sql’ field above. Used by engines to perform necessary translations to the SQL dialect supported by the engine. |
+| Required | default-catalog | A string that specifies the catalog of the user session when the view was created / replaced. Used to resolve the tables used in the view definition. |
```

Comment:
This was brought up on the original proposal as well, these two fields seem a bit specific to certain types of engines. Maybe it would be helpful to me to see some examples from different engines and how views would be represented in them.

A few more questions:
1. what if the catalog is set up differently or with a different name, can these be overriden?
2. why are they required?
3. how are these used at run time. If I have a query `select * from baz` and the `default-catalog=foo` and `default-namespace=bar` are we doing string replacement to then execute `select * from foo.bar.baz` or is it assumed that the user has to run a `USE CATALOG foo;USE NAMESPACE bar` statement first? String replacement sounds iffy as it can't really be done w/o understanding the sql (which we don't want to do yet)

Reply:
1. Catalog setup has to be consistent (e.g., no rename) between view creation and usage.
2. Not required. Fixed. Also change the field names to `session-catalog` and `session-namespace`, because "default" is a little confusing.
3. Fully qualify identifiers in the parsed plan. Don't do string replacement which is iffy as you pointed out.

Reply:
re 3. how do we get "Fully qualify identifiers in the parsed plan. " It still isn't clear to me how these values should be used at runtime.

Reply:
For example, view sql is `select * from tbl1`, session-catalog is `cat1`, and session-namespace is `ns1`. When the view is accessed, after CTE substitution, we will perform view substitution which looks for unresolved relations in the parsed plan for view sql. Here we find `tbl` which does not have namespace and catalog, we will convert it to `cat1.ns1.tbl1`.

Reply:
So an engine has to use these fields in its planning stage? That makes sense, sorry for the misunderstanding. I suppose this is potentially iterative. Eg search for `tbl1` and not found, search for `ns1.tbl1` not found, search for `cat1.ns1.tbl1` found?

Or is it assumed that only iceberg tables are in an iceberg view so the lookup can be done from inside iceberg code w/o engines having to change?

Reply:
No iterative search. If the identifier has 1 part, `tbl` -> `cat1.ns1`; if the identifier has 2 parts, `ns.tbl` -> `cat1.ns.tbl`. Ultimately the purpose of these 2 fields is to help query engines resolve identifiers without ambiguity. Of course, engines can choose to send us sql text with all identifiers qualified, then these 2 fields are optional.